src/jsonpath/json_path.rs

```diff
@@ -0,0 +1,1064 @@
+                        json.get_key(&(s.replace("\\\\", "\\").replace("\\\"", "\"")))
+                    }
+                    Rule::string_value_escape_2 => {
+                        json.get_key(&(s.replace("\\\\", "\\").replace("\\'", "'")))
+                    }
+                    _ => panic!("{}", format!("{:?}", c)),
+                };
+                if let Some(e) = curr_val {
+                    self.calc_internal(pairs.clone(), e, None, calc_data);
+                }
+            }
+        }
+    }
+
+    fn calc_abs_index(&self, i: i64, n: usize) -> usize {
+        if i >= 0 {
+            (i as usize).min(n)
+        } else {
+            (i + n as i64).max(0) as usize
+        }
+    }
+
+    fn calc_indexes<'j: 'i, 'k, 'l, S: SelectValue>(
+        &self,
+        pairs: Pairs<'i, Rule>,
+        curr: Pair<'i, Rule>,
+        json: &'j S,
+        path_tracker: Option<PathTracker<'l, 'k>>,
+        calc_data: &mut PathCalculatorData<'j, S, UPTG::PT>,
+    ) {
+        if json.get_type() != SelectValueType::Array {
+            return;
+        }
+        let n = json.len().unwrap();
+        if let Some(pt) = path_tracker {
+            for c in curr.into_inner() {
+                let i = self.calc_abs_index(c.as_str().parse::<i64>().unwrap(), n);
+                let curr_val = json.get_index(i);
+                if let Some(e) = curr_val {
+                    let new_tracker = Some(create_index_tracker(i, &pt));
+                    self.calc_internal(pairs.clone(), e, new_tracker, calc_data);
+                }
+            }
+        } else {
+            for c in curr.into_inner() {
+                let i = self.calc_abs_index(c.as_str().parse::<i64>().unwrap(), n);
+                let curr_val = json.get_index(i);
+                if let Some(e) = curr_val {
+                    self.calc_internal(pairs.clone(), e, None, calc_data);
+                }
+            }
+        }
+    }
+
+    fn calc_range<'j: 'i, 'k, 'l, S: SelectValue>(
+        &self,
+        pairs: Pairs<'i, Rule>,
+        curr: Pair<'i, Rule>,
+        json: &'j S,
+        path_tracker: Option<PathTracker<'l, 'k>>,
+        calc_data: &mut PathCalculatorData<'j, S, UPTG::PT>,
+    ) {
+        if json.get_type() != SelectValueType::Array {
+            return;
+        }
+        let n = json.len().unwrap();
+        let curr = curr.into_inner().next().unwrap();
+        let (start, end, step) = match curr.as_rule() {
+            Rule::right_range => {
+                let mut curr = curr.into_inner();
+                let start = 0;
+                let end =
+                    self.calc_abs_index(curr.next().unwrap().as_str().parse::<i64>().unwrap(), n);
+                let step = match curr.next() {
+                    Some(s) => s.as_str().parse::<usize>().unwrap(),
+                    None => 1,
+                };
+                (start, end, step)
+            }
+            Rule::all_range => {
```

Comment:
Seems that tests are not considered by codecov? (`array_range_step_all` should cover this, no?)

Reply:
So this is a build system issue no?